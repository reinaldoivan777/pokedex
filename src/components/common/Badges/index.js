import React, { Fragment } from "react";
import { Badge } from "react-bootstrap";

const Badges = ({ datas, variant }) => {
  return (
    <Fragment>
      {datas &&
        datas.map(data => (
          <Badge className="mr-1" key={data} variant={variant}>
            {data}
          </Badge>
        ))}
    </Fragment>
  );
};

export default Badges;
