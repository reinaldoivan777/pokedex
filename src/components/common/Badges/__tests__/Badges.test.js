import React from "react";
import renderer from "react-test-renderer";

import Badges from "../index";

describe("Badges", () => {
  it("render without error", () => {
    renderer.create(<Badges datas={["fire", "electric"]} variant={"info"} />);
  });

  it("empty data", () => {
    renderer.create(<Badges datas={[]} variant={"info"} />);
  });

  it("data null", () => {
    renderer.create(<Badges datas={null} variant={"info"} />);
  });

  it("snapshot renders", () => {
    const component = renderer.create(<Badges datas={["fire", "electric"]} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
