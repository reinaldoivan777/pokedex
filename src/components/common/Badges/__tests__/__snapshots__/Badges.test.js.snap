// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badges snapshot renders 1`] = `
Array [
  <span
    className="mr-1 badge"
  >
    fire
  </span>,
  <span
    className="mr-1 badge"
  >
    electric
  </span>,
]
`;
